pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -1937,45 +1960,46 @@ fn basic_fuzzing() {
                 ));
             }
 
+            // Get state of locked tokens
+            let mut locked_assets = HashMap::new();
+            for user in &users {
+                locked_assets.insert(
+                    user.did,
+                    <PortfolioLockedAssets<TestStorage>>::iter_prefix(
+                        PortfolioId::default_portfolio(user.did),
+                    )
+                    .collect::<Vec<_>>(),
+                );
+            }
+
             next_block();
 
-            for i in 0..40 {
-                for j in 0..4 {
+            for ticker in &tickers {
+                for user in &users {
                     if fail {
                         assert_eq!(
-                            Portfolio::locked_assets(
-                                PortfolioId::default_portfolio(users[j].did),
-                                &tickers[i]
-                            ),
-                            if failed_user == j { 0 } else { 1 }
+                            Settlement::instruction_details(instruction_counter).status,
+                            InstructionStatus::Failed
```

Comment:
This check doesn't need to be inside the for loops.  It doesn't depend on the ticker/user.